### ISSUE TITLE
Notification Popup "Don't show this message again" Is Not Working

### ISSUE THREAD

**Issue by mattmilano** [linuxmint/cinnamon#11894]
### Distribution

LMDE 6

### Package version

5.8.4

### Graphics hardware in use

AMD Ryzen 7 4800H with Radeon Graphics × 8

### Frequency

Always

### Bug description

Clicking "Don't show message again" in the Notification Popup does not work. For example, the notification when connecting or disconnecting from a WiFi network continues to pop up regardless of whether "Don't show this message again" has been pressed.

### Steps to reproduce

- Connect to a network.
- Click "Don't show this message again" in the resulting notification.
- Disconnect from network.
- Reconnect to network and the notification will reappear.

### Expected behavior

Notification should not appear for the specified action.

**Comment by P-D-G** [comment]
I'm having the same issue.

**Comment by jtcraigjr** [comment]
yeah,it is a PITA when you already know that you just did something, and now a Notification comes by and says "you saved a Screenshot"
oh really, must have been my alter ego, I don't remember that  

John in OkieLand...   
 

    On Sunday, October 22, 2023 at 02:11:25 PM CDT, Philip-Dylan Gleonec ***@***.***> wrote:  
 
 


I'm having the same issue.

**Comment by Modiug** [comment]
Same here. I freshly installed LMDE 6 a couple of days ago and I get a WiFi network connected notification whenever my laptop comes out of standby. Clicking "Don't show this message again" has no effect.

**Comment by javimatia** [comment]
Same issue here, it's annoying that so stupid notification..

**Comment by DaveHarton** [comment]
Same, LMDE 6, that same annoying notification every time my PC wakes up and connects to WiFi

**Comment by rutilious** [comment]
same here, every time the computer goes to sleep the message about the network, connected or not connected still continues to appear, I always click on the don't show again button. LMDE6 
6.1.0-18-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.76-1 (2024-02-01) x86_64 GNU/Linux

**Comment by ParticleMon** [comment]
Clicking "Don't show message again" on connect and disconnect notifications apparently should effectively execute the following, respectively, but do not—and evidently have not for over a decade, perhaps since the inception of these buttons:

`gsettings set org.gnome.nm-applet disable-connected-notifications false`
`gsettings set org.gnome.nm-applet disable-disconnected-notifications false`

https://github.com/linuxmint/Cinnamon/issues/2160

**Comment by nicknugget** [comment]
None of the buttons for muting the notifications do what they are supposed to. Neither the one to ignore temporary messages nor the one to switch off notifications completely. The stupid thing is that there are times when you need notifications to acknowledge the pairing of a new Bluetooth device.

Unfortunately, it is also not possible to set which notifications you want and which you don't. The applet under Cinnamon virtually spams you. I am absolutely not interested in reading 200 times a day that my Bluetooth mouse has disconnected and reconnected. Or all the messages for which other applets and programs already generate notifications.

This is much better solved under XFCE.

**Comment by ThePrivacyPlayer** [comment]
Still an issue in May 2024. I had to use the commands shared here to disable it, but hopefully this gets resolved soon, it's been months already and will certainly be extremely frustrating to the newcomers that will switch from windows following their new AI stuff, and when windows 10 comes to end of life next year (I know I'll fully switch at that moment).

**Comment by Secret-chest** [comment]
Can confirm, but this may be a good thing since there's no way to graphically restore them. Running gsettings commands is annoying.

**Comment by ottercynical** [comment]
What kind of actually ridiculous bug is this?  How/why write code for buttons to appear there, but stop short at adding the code to connect them to anything?  Lack of skill/knowledge?

**Comment by ottercynical** [comment]
What kind of bug is this? How/why write code for buttons to appear there, but stop short at adding the code to connect them to anything?

These buttons have never worked, apparently, since they appeared over a decade ago, and have always required either clunky, hacky workarounds with peripheral ramifications or arcane programming sidequests to effectively hack-in basic functionality left abandoned in an unfinished state years prior.

#2160 

Still a bug in 2026 btw, obviously. This will not get fixed, as it is not used in the typical workflow of the average backend programmer, who exclusively uses headless Vim on Apple hardware with all NIC/RF interfaces and pointer devices disabled, and is therefore not a priority to the end-user base that actually uses those things.

**Comment by francocurotto** [comment]
Still not working on LMDE 6

**Comment by the-moog** [comment]
> Clicking "Don't show message again" on connect and disconnect notifications apparently should effectively execute the following, respectively, but do not—and evidently have not for over a decade, perhaps since the inception of these buttons:
> 
> `gsettings set org.gnome.nm-applet disable-connected-notifications true` `gsettings set org.gnome.nm-applet disable-disconnected-notifications true`
> 
> [#2160](https://github.com/linuxmint/cinnamon/issues/2160)

nm-applet (network manager) is only for transient network connections, the annoying messages I get are from the wireless mouse sleeping when not used for more than 10s. On my system nm-applet is not even installed as I am on a wired connection with a fixed IP.

Here, the annoying e.g. mouse disconnect (i.e. sleep) and reconnect (i.e. wake) messages are from the rather dumb Bluetooth `ConnectionNotifier` plugin. Bluetooth->Plugins.
To block the messages, just disable this plugin, but that is a bit of a blunt solution as it will make working with other devices (e.g. a headset) a problem.

Really that plugin is the cause as it should capture the device messages and offer to block them if so configured, storing that filter permanently somehow.  It has little attribution, for me it's author is 'cschramm' and that name takes me to the Blueman project and this bug: https://github.com/blueman-project/blueman/issues/1556
But that descends into a "how to disable notifications" issue and is closed, rather than fixing it.

There is a setting in Settings->Notifications `Ignore transient notifications` that should probably be dealing with this but it does not. 

[This suggestion](https://pupuweb.com/how-block-bluetooth-notifications-specific-devices-linux/) may lead to a work-round, but for me it did not as I could not work out a filter that blocked the message without also blocking the proper device initialisation or possibly affecting other devices. 
Note: To work in user space you need to change --system to --session as well as apparmor stops users listening to system messages.  A system wide filter would not make sense for a user preference. I guess there could be some sort of daemon that lived system wide and filtered all messages based on a database of user preferences hooked to the per-user settings - but that seems nuts to me when it's the plugin that needs fixing.
